tests/src/test/scala/org/http4s/DecodeSpec.scala

```diff
@@ -140,16 +143,21 @@ class DecodeSpec extends Http4sSuite {
     })
   }
 
-  test("decode should either succeed or raise a CharacterCodingException") {
+  test("decode stream result should be consistent with nio's decode on full stream") {
     forAll { (bs: Array[Byte], cs: Charset) =>
+      val referenceDecoder = cs.nioCharset
+        .newDecoder()
+        // setting these to be consistent with our decoder's behavior
+        // note that java.nio.charset.Charset.decode and fs2's utf8Decode
+        // will replace character instead of raising exception
```

Comment:
Maybe this is a behavior worth revisiting, if we're inconsistent with both fs2 and java defaults. :thinking: